### PR TITLE
Azure prow refactor

### PR DESCRIPTION
**Pull request by jim-minter**
Counterpart to https://github.com/openshift/openshift-azure/pull/1507

### CODE REVIEW

File: ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-master.yaml [hunk 1]
```diff
@@ -36,63 +36,4 @@ tag_specification:
   cluster: https://api.ci.openshift.org
   name: origin-v3.11
   namespace: openshift
-tests:
-- as: verify
-  commands: make verify
```
Comment (mjudeikis) [inline ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-master.yaml]: this have to stay

**Comment by jim-minter** [comment]
lgtm'ing this in

**Comment by stevekuznetsov** [comment]
Why?

**Comment by stevekuznetsov** [comment]
Also FWIW the name changes are a bit poor -- the _name_ of the job should be unique across all jobs for everyone's sanity, so something like `pull-ci-org-repo-branch-name` is what we chose. The _context_ of the job is what gets pushed to GitHub anyway, so that can be the short name

**Comment by mjudeikis** [comment]
@stevekuznetsov complexity was growing exponentially on our end. With more test cases piling up and more cloud testing, it became really hard to keep all this up-to-date. Considering we were not able to use prowgen because of multi-dependency problem between our targets. So all our test files were handcrafted.  

Re names: Point taken, will rename those

**Comment by stevekuznetsov** [comment]
Sounds good

**Pull request merged**